### PR TITLE
Bugfix/816 amd bluebird dep

### LINKED ISSUE #816: AMD build broken due to included dependency Bluebird being AMD itself
The 0.11.0 AMD build is broken because of multiple defines. Bluebird is part of the concatenated script that is wrapped in the AMD define but it is in UMD format itself.

Possible solutions:
1) Exclude Bluebird (and any future dependencies) from the AMD build and make it an external dependency (`define(['bluebird'], function(...` in remotestorage.amd.js)
2) Cancatenate a build of Bluebird that has no AMD/UMD wrapper (it's just manipulating globals and does not return/export anything anyway, right?)

Marionette as an example does not include the dependencies Backbone and Underscore. There are two build versions though, one that includes small dependencies closely tied to the project as described in option 2) and one that doesn't (called "core").

### PR DESCRIPTION
**Pull request by silverbucket**
Resolves #816 

Define `bluebird` as an external dependency, update build process to exclude bluebird from AMD builds, added `doc/amd.md` with a basic overview of use.

Ready for review.

### CODE REVIEW

**Comment by silverbucket** [comment]
just a heads up, I know this is failing, will flag is as `ready` and ping everyone when it's ready for review.

**Comment by skddc** [comment]
Cool, thanks!

**Comment by silverbucket** [comment]
ping @skddc @galfert @Ragnis anyone up for review? then we can push out a patch release.

**Comment by skddc** [comment]
+1

File: doc/amd.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#  Using the AMD build of remotestorage.js 
+
+We provide a build of remotestorage.js that works with AMD. External dependencies must be provided for this build of the library to work.
+
+# Release
+
+For every release we provide an AMD build, this file is location in:
```
Comment (skddc) [inline doc/amd.md]: "is located" in.

**Comment by skddc** [comment]
I think we should add a link/comment to the AMD doc to that section of the README.

**Comment by silverbucket** [comment]
@skddc fixed typo and added reference in README.

File: README.md [hunk 2]
```diff
@@ -60,6 +60,9 @@ There are a number of different builds available:
   <kbd>remotestorage-nocache.min.js</kbd> - same as the other .amd/.min
   build, but based on remotestorage-nocache.js.
 
+
+For more information on using the AMD build(s) and it's dependencies, see the [AMD documentation](https://github.com/remotestorage/remotestorage.js/blob/master/doc/amd.md).
```
Comment (galfert) [inline README.md]: s/it's/its

File: doc/amd.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+releases/<version>/remotestorage-nocache.amd.js
+```
+
+The `nocache` build does not include any of the syncing functionality. When a document is saved, or fetched, it is communicating directly with the back-end remotestorage server.
+
+**NOTE** version `0.11.0` of remotestorage.js has a broken AMD build.
+
+# Dependencies
+
+The AMD build of remotestorage.js depends on Bluebird for Promise functionality, so you must include a reference to `bluebird` in your AMD init script.
```
Comment (galfert) [inline doc/amd.md]: Might be a good idea to link to the Bluebird repo here.

**Comment by silverbucket** [comment]
@galfert @skddc ok, link added and grammar corrected

**Comment by galfert** [comment]
+1

**Comment by skddc** [comment]
+1

**Pull request merged**